daemon/cmd/hubble.go

```diff
@@ -137,10 +137,36 @@ func (d *Daemon) launchHubble() {
 		logger.WithFields(logrus.Fields{
 			"address": option.Config.HubbleMetricsServer,
 			"metrics": option.Config.HubbleMetrics,
+			"tls":     option.Config.HubbleMetricsServerTLSEnabled,
 		}).Info("Starting Hubble Metrics server")
 		grpcMetrics := grpc_prometheus.NewServerMetrics()
 
-		if err := metrics.EnableMetrics(log, option.Config.HubbleMetricsServer, option.Config.HubbleMetrics, grpcMetrics, option.Config.EnableHubbleOpenMetrics); err != nil {
+		var metricsTLSConfig *certloader.WatchedServerConfig
+		if option.Config.HubbleMetricsServerTLSEnabled {
+			metricsTLSConfigChan, err := certloader.FutureWatchedServerConfig(
+				logger.WithField("config", "hubble-metrics-server-tls"),
+				option.Config.HubbleMetricsServerTLSClientCAFiles,
+				option.Config.HubbleMetricsServerTLSCertFile,
+				option.Config.HubbleMetricsServerTLSKeyFile,
+			)
+			if err != nil {
+				logger.WithError(err).Error("Failed to initialize Hubble metrics server TLS configuration")
+				return
+			}
+			waitingMsgTimeout := time.After(30 * time.Second)
```

Comment:
Seems like we are blocking all Hubble initialization until TLS config is ready. Would it be possible to move this step further ahead in init process so that other init code can run unaffected by the TLS certs config?

Reply:
I don't think we can really change the order, since the gRPC server requires the metrics to be configured for multiple different reasons. This shouldn't be an issue in practice since the container won't be started until the secret volumes are mounted.

Reply:
Ah right, I forgot they're optional.